### PR TITLE
Upgrade 1.11.0

### PR DESCRIPTION
**Pull request by girazoki**

### CODE REVIEW

**Review by Agusrodri — APPROVED** [review]

File: template/node/src/service.rs [hunk 1]
```diff
@@ -444,7 +470,7 @@ pub fn start_instant_seal_node(config: Configuration) -> Result<TaskManager, sc_
 				transaction_pool: Some(OffchainTransactionPoolFactory::new(
 					transaction_pool.clone(),
 				)),
-				network_provider: network.clone(),
+				network_provider: Arc::new(network.clone()),
```
Comment (tmpolaczyk) [inline template/node/src/service.rs:473]: This looks wrong but I see polkadot-sdk also does it...

File: template/node/src/service.rs [hunk 2]
```diff
@@ -201,12 +202,15 @@ async fn build_relay_chain_interface(
 ///
 /// This is the actual implementation that is abstract over the executor and the runtime api.
 #[sc_tracing::logging::prefix_logs_with("Parachain")]
-async fn start_node_impl(
+async fn start_node_impl<N>(
```
Comment (tmpolaczyk) [inline template/node/src/service.rs:205]: Why does this need a generic? Isn't `N` always `sc_network::NetworkWorker`?

Comment (girazoki) [reply template/node/src/service.rs:205]: not always: https://github.com/paritytech/polkadot-sdk/blob/313fe0f9a277f27a4228634f0fb15a1c3fa21271/substrate/bin/node/cli/src/service.rs#L813

File: template/node/src/service.rs (hunk 1 above)
Comment (girazoki) [reply template/node/src/service.rs:473]: yeah the compiler did not complaint about it, why do you think is wrong?

File: template/node/src/service.rs (hunk 2 above)
Comment (girazoki) [reply template/node/src/service.rs:205]: I did not implement the litep2p network one, but I guess better leave it prepared for it

Comment (tmpolaczyk) [reply template/node/src/service.rs:205]: Ok sounds good

File: template/node/src/service.rs (hunk 1 above)
Comment (tmpolaczyk) [reply template/node/src/service.rs:473]: If `network` is already an `Arc`, then this is like having an `Arc<Arc<T>>`, it works but it doesn't make sense.

**Review by tmpolaczyk — APPROVED** [review]

**Pull request merged**